### PR TITLE
SoftKeywords: allow matching on token string

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala [hunk 1]
```diff
@@ -10,87 +10,83 @@ import scala.meta.tokens.Token.LeftBracket
...
-
   @classifier
   trait KwAs {
     val name = "as"
-    def unapply(token: Token): Boolean = {
-      matches(token, name, dialect.allowAsForImportRename)
-    }
+    @inline def isEnabled = dialect.allowAsForImportRename
+    @inline final def unapply(token: Token): Boolean = isEnabled && name == token.toString
+    @inline final def unapply(token: String): Boolean = isEnabled && name == token
```
Comment (dos65) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala:18]: @kitbellew could you specify why this is needed? Are you going to use `unapply(token: String)`  in scalafmt?

Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala:18]: most immediately, in scalameta, in the next commit. to replace `token.is[soft.A] || token.is[soft.B] || ...`, which invokes `token.toString` multiple times.

Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala:18]: Maybe let's included it in this PR?

Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala:18]: @tgodzik can't include it yet, that change also depends on a commit which _itself_ depends on #2628.

Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala:18]: ok, let me refactor that part out and include it here, give me a minute

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala [hunk 2]
```diff
@@ -196,26 +196,30 @@ class ScannerTokens(tokens: Tokens, input: Input)(implicit dialect: Dialect) {
...
-        ) || (token.is[soft.KwTransparent] && token.next.is[KwTrait]) ||
-        ((token.is[soft.KwOpen] || token.is[soft.KwInfix]) && token.next.is[DefIntro]) ||
-        token.is[InlineMatchMod]
-
+        @inline def nextIsDclIntroOrModifierOr(f: Token => Boolean): Boolean =
+          token.next match {
+            case DclIntro() | Modifier() => true
+            case t => f(t)
+          }
+        token.toString match {
```
Comment (kitbellew) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala:204]: this is where it is used

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala (hunk 1 above)
Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala:18]: done, see other comment

**Review by tgodzik — APPROVED** [review]
LGTM

**Pull request merged**